redash/query_runner/mysql.py

```diff
@@ -140,21 +140,39 @@ def run_query(self, query, user):
                                          ssl=self._get_ssl_parameters())
             cursor = connection.cursor()
             logger.debug("MySQL running query: %s", query)
-            cursor.execute(query)
-
-            data = cursor.fetchall()
-
-            # TODO - very similar to pg.py
-            if cursor.description is not None:
-                columns = self.fetch_columns([(i[0], types_map.get(i[1], None)) for i in cursor.description])
-                rows = [dict(zip((c['name'] for c in columns), row)) for row in data]
-
+            queries = query.rstrip(';').split(';')
+
+            transaction = False
+            transaction_rows = 0
+
+            for query in queries:
+                cursor.execute(query)
+                rows_count = cursor.rowcount
+                data = cursor.fetchall()
+                if cursor.description is not None:
+                    columns = self.fetch_columns([(i[0], types_map.get(i[1], None)) for i in cursor.description])
+                    rows = [dict(zip((c['name'] for c in columns), row)) for row in data]
+                    data = {'columns': columns, 'rows': rows}
+                    json_data = json.dumps(data, cls=JSONEncoder)
+                    break
+                else:
+                    transaction = True
+                    columns = [{'name': 'Row(s) Affected',
+                                'type': types_map.get(str(type(rows_count)).upper(), None)}]
```

Comment:
Wouldn't it be always an integer? In which case you can use `TYPE_INTEGER` constant.